### PR TITLE
distinctUntilChanged based on keyPath

### PR DESCRIPTION
**Pull request by BalazsSzamody**
`distinctUntilChanged` overload where the comparison can be made based on the supplied `keyPath`, works with optionals too (thanks to @Max on RxSwift Slack)

### CODE REVIEW

**Comment by MortyMerr** [comment]
@BalazsSzamody I guess u need to run spm update script

**Comment by BalazsSzamody** [comment]
@MortyMerr Thanks, done

File: RxSwift/Observables/DistinctUntilChanged.swift [hunk 1]
```diff
@@ -61,6 +61,26 @@ extension ObservableType {
         -> Observable<Element> {
             return DistinctUntilChanged(source: self.asObservable(), selector: keySelector, comparer: comparer)
     }
+    
+    /**
+     Returns an observable sequence that contains only distinct contiguous elements based on the keyPath.
+     
+     - parameter keyPath: The keypath for the prioperty that serves as the base for the comparison.
```
Comment (yunuserenguzel) [inline RxSwift/Observables/DistinctUntilChanged.swift]: Fix spelling `property`

Comment (BalazsSzamody) [inline RxSwift/Observables/DistinctUntilChanged.swift]:
```suggestion
     - parameter keyPath: The keypath for the property that serves as the base for the comparison.
```

Comment (BalazsSzamody) [reply RxSwift/Observables/DistinctUntilChanged.swift]: Done. Thanks for the review

**Comment by freak4pc** [comment]
Hey @BalazsSzamody - thanks for the contribution. There are a couple of formatting issues here but the main question I ask myself is whether this is fitting for RxSwift core, or does it fit more into RxSwiftExt. Lets think about this some more before making any changes here.

**Comment by freak4pc** [comment]
I'm closing every PR older than 6 months. If this is still something you wish to pursue / add into RxSwift, please comment and I can reopen it. Thanks! 

I'm probably adding something similar to this in the next version.

**Comment by freak4pc** [comment]
This was added in 1901c7e752e721418e86ddedb0522ee80b76be76

**Pull request closed without merging**